### PR TITLE
javadoc:javadoc could fail under some circumstances

### PR DESCRIPTION
**Pull request by jglick**
Amends #2956 which @kohsuke said caused problems during `release:perform. Picks up https://github.com/mojohaus/extra-enforcer-rules/issues/13.

@reviewbybees

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -39,7 +39,7 @@ for(i = 0; i < buildTypes.size(); i++) {
                                     "MAVEN_OPTS=-Xmx1536m -Xms512m"]) {
                             // Actually run Maven!
                             // -Dmaven.repo.local=… tells Maven to create a subdir in the temporary directory for the local Maven repository
-                            def mvnCmd = "mvn -Pdebug -U clean install javadoc:javadoc ${runTests ? '-Dmaven.test.failure.ignore' : '-DskipTests'} -V -B -Dmaven.repo.local=${pwd tmp: true}/m2repo -s settings-azure.xml -e"
+                            def mvnCmd = "mvn -Pdebug -U javadoc:javadoc clean install ${runTests ? '-Dmaven.test.failure.ignore' : '-DskipTests'} -V -B -Dmaven.repo.local=${pwd tmp: true}/m2repo -s settings-azure.xml -e"
```
Comment (jglick) [inline Jenkinsfile]: Running the `javadoc` goal first seems to reproduce the issue, for whatever reason.

File: pom.xml [hunk 2]
```diff
@@ -723,7 +723,7 @@ THE SOFTWARE.
           <dependency>
             <groupId>org.codehaus.mojo</groupId>
             <artifactId>extra-enforcer-rules</artifactId>
-            <version>1.0-beta-2</version>
+            <version>1.0-beta-6</version>
```
Comment (jglick) [inline pom.xml]: The actual fix. (`1.0-beta-4` minimally)

**Comment by jglick** [comment]
Hmm, getting

```
  Getting remote pull request #2982...
ERROR: {"message":"Not Found","documentation_url":"https://developer.github.com/v3"}
java.io.FileNotFoundException: https://api.github.com/repos/jenkinsci/jenkins/pulls/2982
	at com.squareup.okhttp.internal.huc.HttpURLConnectionImpl.getInputStream(HttpURLConnectionImpl.java:243)
	at com.squareup.okhttp.internal.huc.DelegatingHttpsURLConnection.getInputStream(DelegatingHttpsURLConnection.java:210)
	at com.squareup.okhttp.internal.huc.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:25)
	at org.kohsuke.github.Requester.parse(Requester.java:602)
	at org.kohsuke.github.Requester.parse(Requester.java:584)
	at org.kohsuke.github.Requester._to(Requester.java:264)
Caused: java.io.FileNotFoundException: {"message":"Not Found","documentation_url":"https://developer.github.com/v3"}
	at org.kohsuke.github.Requester.handleApiError(Requester.java:660)
	at org.kohsuke.github.Requester._to(Requester.java:285)
	at org.kohsuke.github.Requester.to(Requester.java:226)
	at org.kohsuke.github.GHRepository.getPullRequest(GHRepository.java:678)
	at org.jenkinsci.plugins.github_branch_source.GitHubSCMSource.doRetrieve(GitHubSCMSource.java:598)
	at org.jenkinsci.plugins.github_branch_source.GitHubSCMSource.retrieve(GitHubSCMSource.java:479)
	at jenkins.scm.api.SCMSource._retrieve(SCMSource.java:355)
	at jenkins.scm.api.SCMSource.fetch(SCMSource.java:309)
	at jenkins.branch.MultiBranchProject$SCMEventListenerImpl.processHeadCreate(MultiBranchProject.java:1300)
	at jenkins.branch.MultiBranchProject$SCMEventListenerImpl.onSCMHeadEvent(MultiBranchProject.java:1134)
	at jenkins.scm.api.SCMHeadEvent$DispatcherImpl.fire(SCMHeadEvent.java:247)
	at jenkins.scm.api.SCMHeadEvent$DispatcherImpl.fire(SCMHeadEvent.java:230)
	at jenkins.scm.api.SCMEvent$Dispatcher.run(SCMEvent.java:481)
	at …
```

File: pom.xml (hunk 2 above)
Comment (kohsuke) [inline pom.xml]: Locally verified that changing this version would unblock the `javadoc:javadoc`.

**Review by kohsuke — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
LGTM though the repro steps are beyond my Maven knowledge

**Pull request merged**